src/posix/platform/radio_spinel.cpp

```diff
@@ -612,37 +615,80 @@ otError RadioSpinel::ParseRadioFrame(otRadioFrame &aFrame, const uint8_t *aBuffe
 
-    // Timestamp is ms + us.
-    unpacked =
-        spinel_datatype_unpack_in_place(aBuffer, aLength,
-                                        SPINEL_DATATYPE_DATA_WLEN_S                              // Frame
-                                                        SPINEL_DATATYPE_INT8_S                   // RSSI
-                                                        SPINEL_DATATYPE_INT8_S                   // Noise Floor
-                                                        SPINEL_DATATYPE_UINT16_S                 // Flags
-                                                        SPINEL_DATATYPE_STRUCT_S(                // PHY-data
-                                                            SPINEL_DATATYPE_UINT8_S              // 802.15.4 channel
-                                                                        SPINEL_DATATYPE_UINT8_S  // 802.15.4 LQI
-                                                                        SPINEL_DATATYPE_UINT32_S // Timestamp (ms).
-                                                                        SPINEL_DATATYPE_UINT16_S // Timestamp (us).
-                                                            ) SPINEL_DATATYPE_STRUCT_S(          // Vendor-data
-                                                            SPINEL_DATATYPE_UINT_PACKED_S        // Receive error
-                                                            ),
-                                        aFrame.mPsdu, &size, &aFrame.mInfo.mRxInfo.mRssi, &noiseFloor, &flags,
-                                        &aFrame.mChannel, &aFrame.mInfo.mRxInfo.mLqi, &msec, &usec, &receiveError);
+#if OPENTHREAD_CONFIG_48BIT_TIMESTAMP_SUPPORT_ENABLE
+    if (mTimestampType == kTimestampType64 || mTimestampType == kTimestampTypeUnknown)
+#endif
+    {
+        unpacked = spinel_datatype_unpack_in_place(
+            aBuffer, aLength,
+            SPINEL_DATATYPE_DATA_WLEN_S                          // Frame
+                            SPINEL_DATATYPE_INT8_S               // RSSI
+                            SPINEL_DATATYPE_INT8_S               // Noise Floor
+                            SPINEL_DATATYPE_UINT16_S             // Flags
+                            SPINEL_DATATYPE_STRUCT_S(            // PHY-data
+                                SPINEL_DATATYPE_UINT8_S          // 802.15.4 channel
+                                        SPINEL_DATATYPE_UINT8_S  // 802.15.4 LQI
+                                        SPINEL_DATATYPE_UINT64_S // Timestamp (us).
+                                ) SPINEL_DATATYPE_STRUCT_S(      // Vendor-data
+                                SPINEL_DATATYPE_UINT_PACKED_S    // Receive error
+                                ),
+            aFrame.mPsdu, &size, &aFrame.mInfo.mRxInfo.mRssi, &noiseFloor, &flags, &aFrame.mChannel,
+            &aFrame.mInfo.mRxInfo.mLqi, &aFrame.mInfo.mRxInfo.mTimestamp, &receiveError);
+    }
+
+#if OPENTHREAD_CONFIG_48BIT_TIMESTAMP_SUPPORT_ENABLE
```

Comment:
Between NCP and `wpantund` (host driver) we enforce the policy that new NCP firmware should remain compatible and work with older driver.

But I think our policy related to RCP and Posix-app was that we require same version on both side. IIRC, in past we allowed changes that would break RCP/Posix-app backward compatibility. Should we keep the policy (i.e. require both side to run same version on RCP/Posix-app model)? If so, do we want/need this feature?

Thoughts?

Reply:
Yes, I agree with you! Just removed this.

Reply:
Thanks.